tests/models/test_main.py

```diff
@@ -565,6 +565,55 @@ def test_query_brief_applications_closed_at_date_for_mix_of_brief_lengths(self):
             db.session.commit()
             assert Brief.query.filter(Brief.applications_closed_at == datetime(2016, 3, 17, 23, 59, 59)).count() == 3
 
+    # TODO: add cases for querying created_at/updated_at auto timestamps with freeze_time
+
+    @pytest.mark.parametrize('inclusive,expected_count', [(True, 2), (False, 1)])
+    @pytest.mark.parametrize('date_format', ["2017-01-01", datetime(2017, 1, 1, 0, 0, 0)])
+    @pytest.mark.parametrize('datestamp_attr', ['published_at', 'withdrawn_at', 'cancelled_at', 'unsuccessful_at'])
+    def test_query_brief_has_date_field_before_and_after(self, datestamp_attr, date_format, inclusive, expected_count):
+        with self.app.app_context():
+            new_briefs = [Brief(data={}, framework=self.framework, lot=self.lot) for i in range(3)]
+            setattr(new_briefs[0], datestamp_attr, datetime(2016, 12, 31, 23, 59, 59))  # < date
+            setattr(new_briefs[1], datestamp_attr, datetime(2017, 1, 1, 0, 0, 0))  # <= and >= date
+            setattr(new_briefs[2], datestamp_attr, datetime(2017, 1, 1, 0, 0, 1))  # > date
```

Comment:
`setattr(new_briefs[2], datestamp_attr, datetime(2017, 1, 1, 0, 0, 1))  # > date` should this instead be
`setattr(new_briefs[2], datestamp_attr, datetime(2017, 1, 2, 0, 0, 0))  # > date`?

What you currently have is 1 minute into the search date, which is still therefore on the same date rather than being > than the search date.